wallet-next/src/storage.rs

```diff
@@ -1,30 +1,84 @@
+use std::path::PathBuf;
+
 use penumbra_chain::params::ChainParams;
 use penumbra_crypto::{
     merkle::{NoteCommitmentTree, Tree},
     note::Commitment,
     FieldExt,
 };
 use penumbra_proto::{crypto::FullViewingKey, Message, Protobuf};
-use sqlx::{query, Pool, Sqlite};
+use sqlx::{migrate::MigrateDatabase, query, Pool, Sqlite};
 
 use crate::sync::ScanResult;
 
+const MAX_MERKLE_CHECKPOINTS_CLIENT: usize = 10;
+
 #[derive(Clone)]
 pub struct Storage {
-    pool: Pool<Sqlite>,
+    storage_path: PathBuf,
 }
 
 impl Storage {
-    pub fn new(pool: Pool<Sqlite>) -> Self {
-        Self { pool }
+    pub fn load(storage_path: PathBuf) -> Self {
+        Self { storage_path }
     }
 
-    pub async fn migrate(self: &Storage) -> anyhow::Result<()> {
-        sqlx::migrate!().run(&self.pool).await.map_err(Into::into)
+    pub async fn initialize(
+        self: &Storage,
+        storage_path: PathBuf,
+        fvk: FullViewingKey,
+        params: ChainParams,
+    ) -> anyhow::Result<()> {
```

Comment:
I think this method should be 
```suggestion
    pub async fn initialize(
        storage_path: PathBuf,
        fvk: FullViewingKey,
        params: ChainParams,
    ) -> anyhow::Result<Self> {
```
, so that the `initialize` method is also a constructor, and there are just two different constructors: one for already-existing data (`load`) and one that initializes (`initialize`).  Then there's no reason to move the `Pool` out of the `Storage` struct.